data_structures/src/data_request.rs

```diff
@@ -343,38 +343,39 @@ pub fn create_tally<RT>(
     dr_output: &DataRequestOutput,
     pkh: PublicKeyHash,
     report: &RadonReport<RT>,
-    reveals: Vec<RevealTransaction>,
-    n_commits: usize,
+    reveal_pkhs: Vec<PublicKeyHash>,
+    commit_pkhs: Vec<PublicKeyHash>,
 ) -> Result<TallyTransaction, failure::Error>
 where
     RT: TypeLike,
 {
     if let Stage::Tally(tally_metadata) = &report.metadata {
         let reveal_reward = dr_output.witness_reward;
+        let n_commits = commit_pkhs.len();
+        let mut slashed_witnesses: HashSet<PublicKeyHash> = commit_pkhs.iter().cloned().collect();
 
         let liars = &tally_metadata.liars;
 
-        if reveals.len() != liars.len() {
+        if reveal_pkhs.len() != liars.len() {
             return Err(TransactionError::MismatchingLiarsNumber {
-                reveals_n: reveals.len(),
+                reveals_n: reveal_pkhs.len(),
```

Comment:
`reveal_pkhs.len()` is called twice in a row. Even if there's no performance penalization for it, I think it could use a binding just like you did for `commits_count` aka `n_commits`.